src/components/table/makeSelectable.js

```diff
@@ -280,6 +289,18 @@ function makeSelectable(BaseTable) {
             this.onSelect(selectedItems, index);
         };
 
+        handleTableBlur = () => {
+            const { focusedIndex } = this.state;
+            if (focusedIndex !== undefined) {
+                // table may get focus back right away in the same tick, in which case we shouldn't clear focus
+                this.blurTimerID = setTimeout(this.clearFocus);
+            }
+        };
```

Comment:
Excellent write-up of the problem statement!

I'm curious about `setTimeout()`. Why not invoke `clearFocus()` synchronously? 

`setState()` is an async method so "setTimeout on setState" seems redundant.

Reply:
Thanks @macrotim !

The problem is this: React synthesizes and bubbles up focus and blur events. Therefore, a blur event that originates anywhere in DOM subtree of the table bubbles to the root of the tree where we attach our `onBlur` handler. For example, as you tab through subcomponents of an item row in the files list (rename button -> more options button -> share button), our root `onBlur` handler gets called for each change in active element. These synthetic blur events are immediately followed by a synthetic focus event that puts the focus back where it is supposed to be, i.e. the next element in tab order.

So in order to normalize (or maybe debounce is a better word) the blur event, we need the `setTimeout()`. I did not know about `setState()` being async, but I tried and it doesn't work correctly without the `setTimeout()`. Perhaps it has to do with the order in which React fires synthesized events.

This solution was inspired by this blog post:
https://medium.com/@jessebeach/dealing-with-focus-and-blur-in-a-composite-widget-in-react-90d3c3b49a9b

Reply:
Thanks for the background info! Super helpful!